File: app/frontend/app/models/tag.js
import Ember from 'ember';
import DS from 'ember-data';
import CoughDrop from '../app';
import speecher from '../utils/speecher';
import persistence from '../utils/persistence';
import Utils from '../utils/misc';

CoughDrop.Tag = DS.Model.extend({
  button: DS.attr('raw'),
  tag_id: DS.attr('string'),
  label: DS.attr('string'),
  public: DS.attr('boolean'),
});

export default CoughDrop.Tag;

File: lib/json_api/tag.rb
module JsonApi::Tag
  extend JsonApi::Json
  
  TYPE_KEY = 'tag'
  DEFAULT_PAGE = 10
  MAX_PAGE = 25
    
  def self.build_json(tag, args={})
    json = {}
    
    json['id'] = tag.global_id
    json['tag_id'] = tag.tag_id
    json['public'] = tag.public
    if tag.data['button']
      json['button'] = tag.data['button']
    else
      json['label'] = tag.data['label']
    end
    json
  end
end
